src/react-components/src/components/01-atoms/Forms/DateRangeForm.tsx

```diff
@@ -0,0 +1,91 @@
+/* Input fields for date ranges. */
+import React from "react";
+import Button, { ButtonProps } from "../Button/Button";
+import HelperErrorText, { HelperErrorTextOptions } from "./HelperErrorText";
+import Input, { InputOptions } from "./Input";
+import { LabelOptions } from "./Label";
+import bem from "../../../utils/bem";
+
+export interface DateCheckFormProps {
+  /** Used for BEM css convention. */
+  blockName?: string;
+  /** Used for BEM css convention. */
+  baseClass?: string;
+
+  /** The Label to describe the form */
+  formLabel: JSX.Element;
+
+  /** The Label to describe the from input */
+  fromLabelOpts: LabelOptions;
+  /** The from input */
+  fromInputOpts: InputOptions;
+  /** Helper text for the from input */
+  fromHelper?: HelperErrorTextOptions;
+
+  /** The Label to describe the to input */
+  toLabelOpts: LabelOptions;
+  /** The to input */
+  toInputOpts: InputOptions;
+  /** Helper text for the to input */
+  toHelper?: HelperErrorTextOptions;
+
+  /** Whether the form is currently in an error state */
+  showError: boolean;
+  /** The error text to render */
+  error?: HelperErrorTextOptions;
+
+  /** The submit button */
+  buttonProps?: ButtonProps;
+}
+
+export default function DateRangeForm(props: DateCheckFormProps) {
+  const {
+    blockName = "date-range",
+    baseClass,
+    formLabel,
+    fromLabelOpts,
+    fromInputOpts,
+    fromHelper,
+    toLabelOpts,
+    toInputOpts,
+    toHelper,
+    showError,
+    error,
+    buttonProps } = props;
+    if (showError && !error) {
+      throw new Error("Error should be shown, but no Error Content exists");
+    }
+
+  return (
+    <fieldset className={bem(blockName, [], baseClass)}>
+      <legend>{formLabel}</legend>
+      <div className="input-row">
+        <Input
```

Comment:
I think this is fine for a first pass, but you should have some validation for the input fields (like number type and max 4 digits) so users enter reasonable years.
<img width="213" alt="Screen Shot 2020-04-12 at 5 22 53 PM" src="https://user-images.githubusercontent.com/1280564/79080372-4c227080-7ce2-11ea-9d79-dff76aa39d49.png">